Build warning in hfp_create_connection

**Describe the bug**

Using arm gcc version 12.2, I get the following build warning. It appears to be caused by not checking for NULL.

$ gcc --version

gcc (Ubuntu 12.1.0-2ubuntu1~22.04) 12.1.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[ 76%] Built target kitchen_sink_lwip_background
[ 76%] Building C object test/kitchen_sink/CMakeFiles/kitchen_sink_lwip_poll.dir/__/__/lib/btstack/src/classic/hfp.c.obj
/home/peterh/source/pico/pico-sdk2/lib/btstack/src/classic/hfp.c: In function 'hfp_create_connection.part.0':
/home/peterh/source/pico/pico-sdk2/lib/btstack/src/classic/hfp.c:634:11: error: 'memcpy' offset [0, 5] is out of the bounds [0, 0] [-Werror=array-bounds]
  634 |     (void)memcpy(hfp_connection->remote_addr, bd_addr, 6);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [test/kitchen_sink/CMakeFiles/kitchen_sink_lwip_poll.dir/build.make:3215: test/kitchen_sink/CMakeFiles/kitchen_sink_lwip_poll.dir/__/__/lib/btstack/src/classic/hfp.c.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:2150: test/kitchen_sink/CMakeFiles/kitchen_sink_lwip_poll.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

**To Reproduce**

Build the pico-sdk

--- a/src/classic/hfp.c
+++ b/src/classic/hfp.c
@@ -631,6 +631,9 @@ static hfp_connection_t * hfp_create_connection(bd_addr_t bd_addr, hfp_role_t lo
     hfp_connection_t * hfp_connection = get_hfp_connection_context_for_bd_addr(bd_addr, local_role);
     if (hfp_connection) return  hfp_connection;
     hfp_connection = create_hfp_connection_context();
+    if (!hfp_connection) {
+        return NULL;
+    }
     (void)memcpy(hfp_connection->remote_addr, bd_addr, 6);
     hfp_connection->local_role = local_role;
     log_info("Create HFP context %p: role %u, addr %s", hfp_connection, local_role, bd_addr_to_str(bd_addr));

